src/Modules/SimplCommerce.Module.Inventory/Controllers/WarehouseApiController.cs

```diff
@@ -29,5 +35,174 @@ public WarehouseApiController(IRepository<Warehouse> warehouseRepository)
 
             return Ok(warehouses);
         }
+
+        [HttpPost("grid")]
+        public IActionResult List([FromBody] SmartTableParam param)
+        {
+            var query = _warehouseRepository.Query();
+            if (param.Search.PredicateObject != null)
+            {
+                dynamic search = param.Search.PredicateObject;
+
+                if (search.Name != null)
+                {
+                    string name = $"%{search.Name}%";
+                    query = query.Where(x => EF.Functions.Like(x.Name, name));
+                }
+            }
+
+            var warehouses = query.ToSmartTableResult(
+                param,
+                 sp => new
+                 {
+                     sp.Id,
+                     sp.Name
+                 });
+
+            return Json(warehouses);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get(long id)
+        {
+            var warehouse = await _warehouseRepository.Query().Include(w => w.Address).FirstOrDefaultAsync(w => w.Id == id);
```

Comment:
Humm...when I remove it, it's null within the select. Let me know if you get a different result if you were to make this change locally.

Reply:
Ah, I am sorry, because here you have called FirstOrDefaultAsync. that mean the query has been executed, then map to the AddressVm later.  So .Include(w => w.Address) is the correct usage in this case